src/go/rpk/pkg/api/admin/api_user.go

```diff
@@ -51,6 +51,23 @@ func (a *AdminAPI) CreateUser(ctx context.Context, username, password, mechanism
 	return a.sendToLeader(ctx, http.MethodPost, usersEndpoint, u, nil)
 }
 
+// UpdateUser updates a user with the given username and password using the
+// given mechanism (SCRAM-SHA-256, SCRAM-SHA-512).
+func (a *AdminAPI) UpdateUser(ctx context.Context, username, password, mechanism string) error {
+	if username == "" {
+		return errors.New("invalid empty username")
+	}
+	if password == "" {
+		return errors.New("invalid empty password")
+	}
+	u := newUser{
+		User:      username,
+		Password:  password,
+		Algorithm: mechanism,
```

Comment:
Is it valid to send the PUT request with an empty algorithm?

Reply:
in an empty algorithm?

Reply:
Oh I see, I think the CreateUser was coded thinking in the CLI command, the CLI command has a default for the mechanism so the request will never fail :thinking: 

I'm fine leaving it like this, but if the request **needs** the mechanism, we should fail instead.

Reply:
I just checked:

```
 curl -X PUT localhost:9644/v1/security/users/foo \
  -d '{"password":"bar"}'
{"message": "String algo missing", "code": 400}
```

I think it's better if we control the message.

Reply:
It's likely safe to default this to SCRAM-SHA-256 if it is empty -- it's the same thing we do in rpk since that's what people want ~100% of the time.

Reply:
The issue here is that create does not error if it is empty.... So i am wondering if I can get myself into trouble since the put message may error out. This is further complicated since if the default behavior is already in the post endpoint that i can never be sure what to send in the put message....

Reply:
OK, added a, what i hope is, useful error. Let me know if you approve.